### PR TITLE
Switch schema format to SQL

### PR DESCRIPTION
**Pull request by timhabermaas**
This gets rid of the schema monkey patching.

I've also added a `config/database.yml.travis` file which is used by travis.

### CODE REVIEW

**Comment by timhabermaas** [comment]
Sadly the pre-commit hook fails because `structure.sql` contains an invalid string. This file comes straight from `bundle exec rake db:structure:dump` and I've not modified it. In order to get travis passing I've "fixed" this in 06f5b85. @jfly, could you run `bundle exec rake db:structure:dump` locally and verify that the pre-commit hook fails? I hope that's just my local mysql version and not a general problem.

File: WcaOnRails/config/database.yml.travis [hunk 1]
```diff
@@ -0,0 +1,9 @@
+test:
```
Comment (jfly) [inline WcaOnRails/config/database.yml.travis]: This file doesn't look any different than the existing `WcaOnRails/config/database.yml`, why do we need it?

**Comment by jfly** [comment]
This looks pretty good to me! I do see a tab inside `db/structure.sql` after running `bundle exec rake db:structure:dump`. What a bummer. Instead of having to edit the file by hand, how about adding `db/structure.sql` to `.pre_commit.ignore`?

File: WcaOnRails/app/models/competition.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
 class Competition < ActiveRecord::Base
   self.table_name = "Competitions"
+  # FIXME Tests fail with "Unknown primary key for table Competitions in model Competition."
+  #       when not setting the primary key explicitly. I have
+  #       no clue why... (th, 2015-09-19)
+  self.primary_key = "id"
```
Comment (jfly) [inline WcaOnRails/app/models/competition.rb]: That's interesting. I would not have expected your changes to affect our runtime at all.

Could you add a comment to https://github.com/cubing/worldcubeassociation.org/issues/91 so we remember to get to this?

**Comment by timhabermaas** [comment]
> Instead of having to edit the file by hand, how about adding `db/structure.sql` to `.pre_commit.ignore`?

Oh! I didn't know such a file exists. That's much better!

File: WcaOnRails/config/database.yml.travis (hunk 1 above)
Comment (timhabermaas) [inline WcaOnRails/config/database.yml.travis]: These two files being identical is just a coincidence. I don't want travis to break if we change the local connection settings. Speaking of local settings: I usually don't commit `database.yml` to the repository, but provide a `database.yml.sample` since people might want to customize their database settings which gets annoying if the file you modify is checked into source control. I decided against removing it, though since it would probably break the vagrant box.

Comment (jfly) [inline WcaOnRails/config/database.yml.travis]: Hmm... I think the DATABASE_URL environment variable is a better way of
providing the developer flexibility you're describing. I generally dislike
.sample files, as they're an extra step when setting up an initial
development environment.

I feel like it would be cleaner to use erb inside the .yml file to have
different settings for Travis. Copying a file around seems hacky to me.
What do you think?

**Comment by timhabermaas** [comment]
@jfly I've removed the `database.yml.travis` file and replaced it with `DATABASE_URL` as you suggested. I wasn't aware that `DATABASE_URL` is supported by Rails (4?). That's pretty cool. I used to manually put things together using `database.yml.erb`.

**Comment by timhabermaas** [comment]
tl;dr: Ready for merge.

**Comment by jfly** [comment]
I love it! Merging.
